File: cpp/program/aset.cpp
/*
 * LICENSE: All Rights Reserved
 * (c) 2025 Willsoon Tulus P. Simanjuntak
 * (c) 2025 Universitas Pendidikan Indonesia
 *
 * Saya Willsoon Tulus Parluhutan Simanjuntak mengerjakan evaluasi
 * Tugas Praktikum 3 dalam mata kuliah Desain Pemrograman
 * Berorientasi Objek untuk keberkahanNya maka saya tidak
 * melakukan kecurangan seperti yang telah dispesifikasikan.
 * Aamiin.
 */

/* Deklarasi header custom */
#include "header.hpp"

/* Deklarasi kelas */

// Deklarasi konstruktor awal untuk inisialisasi
Aset::Aset() {
    this->kode = "";
    this->nama = "";
    this->jenis = "";
    this->jumlah = 0;
};

// Konstruktor untuk membuat objek baru oleh fungsi luar
Aset::Aset(string kode, string nama, string jenis, int jumlah) {
    this->kode = kode;
    this->nama = nama;
    this->jenis = jenis;
    this->jumlah = jumlah;
};

// Fungsi setter untuk menentukan kode aset
void Aset::setKode(string kode) {
    this->kode = kode;
};

// Fungsi setter untuk menentukan nama aset
void Aset::setNama(string nama) {
    this->nama = nama;
};

// Fungsi setter untuk menentukan jenis aset
void Aset::setJenis(string jenis) {
    this->jenis = jenis;
};

// Fungsi setter untuk menentukan jumlah aset
void Aset::setJumlah(int jumlah) {
    this->jumlah = jumlah;
};

// Fungsi getter untuk mendapatkan kode aset
string Aset::getKode() {
    return this->kode;
};

// Fungsi getter untuk mendapatkan nama aset
string Aset::getNama() {
    return this->nama;
};

// Fungsi getter untuk mendapatkan jenis aset
string Aset::getJenis() {
    return this->jenis;
};

// Fungsi getter untuk mendapatkan jumlah dari aset yang tersedia
int Aset::getJumlah() {
    return this->jumlah;
};

// Deklarasi destruktor
Aset::~Aset() {};

File: python/program/aset.py
# LICENSE: All Rights Reserved
# (c) 2025 Willsoon Tulus P. Simanjuntak
# (c) 2025 Universitas Pendidikan Indonesia
#
# Saya Willsoon Tulus Parluhutan Simanjuntak mengerjakan evaluasi
# Tugas Praktikum 3 dalam mata kuliah Desain Pemrograman
# Berorientasi Objek untuk keberkahanNya maka saya tidak
# melakukan kecurangan seperti yang telah dispesifikasikan.
# Aamiin.

# Deklarasi kelas objek #

class Aset:

    # constructor untuk inisialisasi atribut

    def __init__(self, kode: str, nama: str, jenis: str, jumlah: int) -> None:
        self._kode: str = kode
        self._nama: str = nama
        self._jenis: str = jenis
        self._jumlah: int = jumlah

    # Fungsi setter (write atribut)
    def setKode(self, kode: str) -> None:
        self._kode = str(kode)

    def setNama(self, nama: str) -> None:
        self._nama = str(nama)

    def setJenis(self, jenis: str) -> None:
        self._jenis = str(jenis)

    def setJumlah(self, jumlah: int) -> None:
        self._jumlah = int(jumlah)

    # Fungsi getter (read atribut)
    def getKode(self) -> str:
        return self._kode

    def getNama(self) -> str:
        return self._nama

    def getJenis(self) -> str:
        return self._jenis

    def getJumlah(self) -> int:
        return self._jumlah
